src/Microsoft.Health.Blob/Features/Health/BlobHealthCheck.cs

```diff
@@ -31,30 +36,46 @@ public class BlobHealthCheck : IHealthCheck
     /// <param name="client">The cloud blob client factory.</param>
     /// <param name="namedBlobContainerConfigurationAccessor">The IOptions accessor to get a named container configuration version.</param>
     /// <param name="containerConfigurationName">Name to get corresponding container configuration.</param>
-    /// <param name="testProvider">The test provider.</param>
+    /// <param name="testProvider">The blob test provider.</param>
+    /// <param name="customerKeyHealthCache">The status of the customer-managed key</param>
     /// <param name="logger">The logger.</param>
     public BlobHealthCheck(
         BlobServiceClient client,
         IOptionsSnapshot<BlobContainerConfiguration> namedBlobContainerConfigurationAccessor,
         string containerConfigurationName,
         IBlobClientTestProvider testProvider,
+        AsyncData<CustomerKeyHealth> customerKeyHealthCache,
```

Comment:
Looks like nobody else uses the BlobHealthCheck, but it is an issue for SqlHealthCheck. I added the registration for the CMK background service to the SQL registration. If CMK options are empty, it will skip keyvault check and always return healthy